mlir/lib/Target/LLVMIR/Dialect/OpenMP/OpenMPToLLVMIRTranslation.cpp

```diff
@@ -2459,6 +2459,8 @@ convertDeclareTargetAttr(Operation *op, mlir::omp::DeclareTargetAttr attribute,
       if (declareType == omp::DeclareTargetDeviceType::host) {
         llvm::Function *llvmFunc =
             moduleTranslation.lookupFunction(funcOp.getName());
+        llvmFunc->replaceAllUsesWith(
+            llvm::UndefValue::get(llvmFunc->getType()));
```

Comment:
@skatrak I have add this line to replace the uses of the function that are to be removed with `undef` values. This should prevent the crash.

Let me know if this looks like an appropriate fix for the issue to you.

Reply:
This looks like a good solution to the problem. Thanks!